### PR TITLE
[POU-187] Watchlist basis

### PR DESCRIPTION
**Pull request by crimson-med**
[POU 187](https://cadtrust.atlassian.net/browse/POU-187)

# Why

Adding the watchlist page

# What

- Adding Watchlist Page
- Adding Mock watchlist data
- Adding watchlist to overmind

### CODE REVIEW

File: src/@types/Watchlist.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export type Watchlist = {
+  name: string
+  description: string
+  id: string
+  projects: number
```
Comment (morphet81) [inline src/@types/Watchlist.tsx:5]: Why is `projects` a number? Shouldn't it be a list of IDs (which are strings)

File: src/@types/Watchlist.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export type Watchlist = {
+  name: string
+  description: string
+  id: string
```
Comment (morphet81) [inline src/@types/Watchlist.tsx:4]: Can we have `id` on top?

File: src/@types/Watchlist.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  projects: number
+  createdAt: string
+}
+export enum WatchlistSorting {
+  RECENTLY_ADDED = `recentlyAdded`,
+  ALPHABETICAL = `alphabetical`,
+  NUMBER_OF_PROJECTS = `numberOfProjects`,
```
Comment (morphet81) [inline src/@types/Watchlist.tsx:11]: Is there a reason for values to not match the casing of enum names?

File: src/app/watchlist/page.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    } else {
+      return renderList()
+    }
+  }
+
+  const handleSort = (selectedSort: Item) => {
+    if (!watchlists) {
+      return
+    }
+    const orderedWatchlists: Watchlist[] = [...watchlists]
```
Comment (morphet81) [inline src/app/watchlist/page.tsx:58]: Break lines before and after

File: src/components/atoms/WatchlistSummary/WatchlistSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Watchlist } from '@/@types/Watchlist'
+import { Card, Text, Heading, CardBody, Flex } from '@chakra-ui/react'
+import { ImportantText } from '@/components/atoms/ImportantText/ImportantText'
+import { Aeonik, AeonikFono } from '@/styles/fonts'
+import { useTranslation } from 'react-i18next'
+
+interface WatchlistSummaryProps {
+  watchlist: Watchlist
+}
```
Comment (morphet81) [inline src/components/atoms/WatchlistSummary/WatchlistSummary.tsx:9]: break a line

**Review by morphet81 — CHANGES_REQUESTED** [review]

File: src/@types/Watchlist.tsx (hunk 1 above)
Comment (crimson-med) [reply src/@types/Watchlist.tsx:5]: This here is only used for displaying the number of projects on the page summary.
When we display the projects on the details page projects will need to be paginated so we will need a separate endpoint to handle the projects as part of watchlist.

Thus we don't need to have the id of the projects.

File: src/@types/Watchlist.tsx (hunk 3 above)
Comment (crimson-med) [reply src/@types/Watchlist.tsx:11]: This was as the value is used in the translation so I used the same format that is used in the translation. Any preferences on how to handle this case?

File: src/@types/Watchlist.tsx (hunk 1 above)
Comment (morphet81) [reply src/@types/Watchlist.tsx:5]: In that case you should rename the the variable to make it clear.

**Review by morphet81 — APPROVED** [review]

**Pull request merged**